subject.py

```diff
@@ -0,0 +1,166 @@
+import json
+from dataclasses import dataclass, field
+from datetime import date
+from recording import RecordingInterface
+from typing import List
+
+
+@dataclass
+class Round:
+    """
+    Holds the information about the game round
+
+    This class holds information about current round and all potential rounds
+    """
+
+    allowed_stages: List = field(default_factory=lambda: ['Baseline', 'Game', 'SingleChoiceGame', 'Stimulation'])
+    current_round: int = 1 or str
+    iap: str = '1'
+    max_rounds: int = 81
+    stage: str = 'Game'
+
+    def increment(self) -> None:
+        """
+        Increment the round by 1 or reset to 1 if we hit the limit
+
+        Returns:
+            None
+        """
+        self.current_round = self.current_round + 1
+
+    def set_stage(self, stage: str) -> None:
+        """
+        Used to set the stage operation
+
+        Args:
+            stage (str): The new stage to set
+
+        Returns:
+            None
+        """
+
+        if stage not in self.allowed_stages:
+            raise ValueError(f'"{stage}" is not an accepted stage for this game')
+
+        self.stage = stage
+
+
+@dataclass
+class Subject:
+    """
+    The test subject
+
+    Holds information about the subject, the date the test occurred and holds
+    the data for the test
+    """
+
+    id: str
+    date: str
+    data: dict
+
+    def get_key(self):
+        """
+        Gets the key for the subject file name
+
+        Returns:
+            str: The subject key
+        """
+
+        return f'{self.id}({self.date})'
+
+    def create_data_template(self):
+        """
+        Create the data template ready for data population
+
+        Returns:
+            dict: The template for the json data
+        """
+
+        self.data = {
+            self.id: {
+                self.date: {
+                    'Timestamps': {
+                        'Baseline': {},
+                        'Stimulation': {}
+                    }
+                }
+            }
+        }
+
+    def add_timestamp(self, current_round: Round, words, stage, timestamp: str) -> None:
+        """
+        Stores the timestamp for a round against a round
+
+        Args:
+            current_round (Round): The round object
+            timestamp (str): The timestamp to store against said round
+
+        Returns:
+            None
+        """
+        if stage == 'Game':
```

Comment:
I would say this could be greatly simplified like such

**Version 1**
```python
game_stages = ['Game', 'SingleChoiceGame']
if current_round.stage in game_stages:
    round_name = f'Q{current_round.current_round}'
    self.data[self.id][self.date]['Timestamps']['Baseline'].update({round_name: timestamp})
    return None

round_name = f'IAPS{current_round.iap}' if current_round.stage == 'Stimulation' else f'Baseline{current_round.iap}'
self.data[self.id][self.date]['Timestamps']['Stimulation'].update({round_name: timestamp})
```

If you really brushed up how Round and Instance were communicating, you could just use the if statements to set the round name and then add it to data

**Version 2** (Better in my opinion)
```python
# Be a good idea to make these class properties
game_stages = ['Game', 'SingleChoiceGame']
round_prefixes = {
    'Stimulation': 'IAPS',
    'Baseline': 'Baseline',
    'Game': 'Q',
    'SingleChoiceGame': 'Q'
}

prefix = round_prefixes[current_round.stage]
round_number = current_round.current_round if current.round.stage in game_stages else current_round.iap
round_name = f'{prefix}{round_number}'
    
self.data[self.id][self.date]['Timestamps'][current_round.stage].update({round_name: timestamp})
```

It's also worth nothing that `pass` isn't needed if you're doing an `elif`.

Reply:
This seems good, if we can make the Round.stage logic work again it should be good. Unfortunately I could not do that

Reply:
Actually, we could improve this greatly by separating a bit of logic out, then the class method would just be

```python
    def add_timestamp(self, current_round: Round, timestamp: str) -> None:
        """
        Stores the timestamp for a round against a round

        Args:
            current_round (Round): The round object
            timestamp (str): The timestamp to store against said round

        Returns:
            None
        """
        
        round_name = current_round.get_round_name()
        self.data[self.id][self.date]['Timestamps'][current_round.stage].update({round_name: timestamp})
```

Reply:
I can write the new get_round name, if you want, since I wrote the logic for the IAPS number and the rest is straight forward

Reply:
Sure thing, I probably won't be able to contribute code updates until the week after next. I've got a lot to do at home, places to be, and a wedding to go away for.

Reply:
Of course, you have said that you were unavailable